src/valhalla_path_comparison.cc

```diff
@@ -106,15 +235,26 @@ int main(int argc, char *argv[]) {
     } catch (...) {
       throw std::runtime_error("No edge/node costing provided");
     }
+  } else if (vm.count("shape")) {
+    map_match = false;
   } else {
-    std::cerr << "The json parameter was not supplied but is required.\n\n" << options << std::endl;
+    std::cerr << "The json parameter or shape parameter was not supplied but is required.\n\n" << options << std::endl;
     return EXIT_FAILURE;
   }
 
   //parse the config
   boost::property_tree::ptree pt;
   boost::property_tree::read_json(config.c_str(), pt);
 
+  // Get something we can use to fetch tiles
+  valhalla::baldr::GraphReader reader(pt.get_child("mjolnir"));
+
+  // If a shape is entered use edge walking
+  if (!map_match) {
```

Comment:
indeed if we just used the `actor` object we just need to setup the request properly based on the options passed in instead of having a separate function. this program gets a lot small and duplicates a lot less of the service level code